sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/converter/KotlinTree.kt

```diff
@@ -30,47 +30,38 @@ import org.sonar.api.batch.fs.InputFile
 import org.sonarsource.kotlin.api.ParseException
 import org.sonarsource.kotlin.converter.KotlinTextRanges.textPointerAtOffset
 
-class KotlinTree private constructor(
-    val psiFile: KtFile,
-    val document: Document,
-    val bindingContext: BindingContext,
-) {
-    companion object {
-        @JvmStatic
-        fun of(content: String, environment: Environment, inputFile: InputFile): KotlinTree {
-            val psiFile: KtFile = environment.ktPsiFactory.createFile(normalizeEol(content))
-            val document = try {
-                psiFile.viewProvider.document ?: throw ParseException("Cannot extract document")
-            } catch (e: AssertionError) {
-                // A KotlinLexerException may occur when attempting to read invalid files
-                throw ParseException("Cannot correctly map AST with a null Document object")
-            }
+class KotlinTree(val psiFile: KtFile, val document: Document, val bindingContext: BindingContext)
 
-            checkParsingErrors(psiFile, document, inputFile)
-            val bindingContext = bindingContext(
-                environment.env,
-                environment.classpath,
-                listOf(psiFile),
-            )
-
-            return KotlinTree(psiFile, document, bindingContext)
-        }
+data class KotlinSyntaxStructure(val ktFile: KtFile, val document: Document, val inputFile: InputFile)
 
-        private fun descendants(element: PsiElement): Sequence<PsiElement> {
-            return element.children.asSequence().flatMap { tree: PsiElement -> sequenceOf(tree) + descendants(tree) }
-        }
+fun generateKtFile(content: String, environment: Environment, inputFile: InputFile): KotlinSyntaxStructure{
+    val psiFile: KtFile = environment.ktPsiFactory.createFile(normalizeEol(content))
+    
+    val document = try {
+        psiFile.viewProvider.document ?: throw ParseException("Cannot extract document")
+    } catch (e: AssertionError) {
+        // A KotlinLexerException may occur when attempting to read invalid files
+        throw ParseException("Cannot correctly map AST with a null Document object")
+    }
+    
+    checkParsingErrors(psiFile, document, inputFile)
+    
+    return KotlinSyntaxStructure(psiFile, document, inputFile)
+}
```

Comment:
To me this looks like a factory method - is there a reason why it shouldn't be inside of the `KotlinSyntaxStructure` class / its companion object to associate it with the type it actually produces?